File: apps/api/app/models/appointment.rb
class Appointment < ApplicationRecord
  # Associations
  belongs_to :onboarding_session
  belongs_to :student
  # TODO: belongs_to :therapist (when therapist model exists)

  # Enums
  enum :status, {
    scheduled: 'scheduled',
    confirmed: 'confirmed',
    completed: 'completed',
    cancelled: 'cancelled',
    no_show: 'no_show'
  }, prefix: true

  # Validations
  validates :therapist_id, presence: true
  validates :scheduled_at, presence: true
  validates :duration_minutes, presence: true, numericality: { only_integer: true, greater_than: 0 }
  validates :status, presence: true
  validate :scheduled_in_future, on: :create

  # Scopes
  scope :upcoming, -> { where('scheduled_at > ?', Time.current).where(status: ['scheduled', 'confirmed']) }
  scope :past, -> { where('scheduled_at < ?', Time.current) }
  scope :for_therapist, ->(therapist_id) { where(therapist_id: therapist_id) }

  # Instance methods
  def end_time
    scheduled_at + duration_minutes.minutes
  end

  def can_cancel?
    status_scheduled? || status_confirmed?
  end

  def can_reschedule?
    status_scheduled? || status_confirmed?
  end

  private

  def scheduled_in_future
    if scheduled_at.present? && scheduled_at < Time.current
      errors.add(:scheduled_at, "must be in the future")
    end
  end
end

File: packages/shared-types/src/appointment.ts
import { z } from 'zod'

export const AppointmentStatusSchema = z.enum([
  'scheduled',
  'confirmed',
  'completed',
  'cancelled',
  'no_show',
])

export type AppointmentStatus = z.infer<typeof AppointmentStatusSchema>

export const AppointmentSchema = z.object({
  id: z.string().uuid(),
  sessionId: z.string().uuid(),
  studentId: z.string().uuid(),
  therapistId: z.string().uuid(),
  scheduledAt: z.date(),
  duration: z.number().int(), // minutes
  status: AppointmentStatusSchema,
  notes: z.string().optional(),
  createdAt: z.date(),
  updatedAt: z.date(),
})

export type Appointment = z.infer<typeof AppointmentSchema>

